pkg/opavalidation/opavalidation.go

```diff
@@ -0,0 +1,427 @@
+// Validation includes signatures for Insights-provided rego functions.
+func ValidateRego(ctx context.Context, regoAsString string, objectAsBytes []byte, eventType string, objectNamespaceOverride string) (actionItems, error) {
+	if !strings.Contains(regoAsString, "package fairwinds") {
+		return nil, errors.New("policy must be within a fairwinds package. The policy must contain the statement: package fairwinds")
+	}
+	objectAsMap, err := objectBytesToMap(objectAsBytes)
+	if err != nil {
+		return nil, err
+	}
+	err = updateObjectWithNamespaceOverride(objectAsMap, objectNamespaceOverride)
+	if err != nil {
+		return nil, fmt.Errorf("while overriding object namespace with %q: %v", objectNamespaceOverride, err)
+	}
+	regoResult, err := runRegoForObject(ctx, regoAsString, objectAsMap)
+	if err != nil {
+		return nil, err
+	}
+	actionItems, err := actionItemsFromRegoResult(regoResult)
+	if err != nil {
+		return actionItems, err
+	}
+	err = updateActionItemsWithObjectFields(actionItems, objectAsMap)
+	if err != nil {
+		return actionItems, err
+	}
+	updateActionItemsWithEventType(actionItems, eventType)
+	return actionItems, nil
+}
+
+// runRegoForObject executes rego with a Kubernetes object as input.
+func runRegoForObject(ctx context.Context, regoAsString string, object map[string]interface{}) (rego.ResultSet, error) {
+	query, err := rego.New(rego.EnablePrintStatements(true), rego.PrintHook(topdown.NewPrintHook(os.Stdout)),
+		rego.Query("results = data"),
+		rego.Module("fairwinds", regoAsString),
+		rego.Function2(
+			&rego.Function{
+				Name: "kubernetes",
+				Decl: types.NewFunction(types.Args(types.S, types.S), types.A),
+			},
+			func(_ rego.BuiltinContext, _ *ast.Term, _ *ast.Term) (*ast.Term, error) {
+				// Perhaps do something mroe here to communicate it isn't possible to fetch
+				// cluster resources?
+				return nil, nil
+			},
+		),
+		rego.Function1(
+			&rego.Function{
+				Name: "insightsinfo",
+				Decl: types.NewFunction(types.Args(types.S), types.A),
+			},
+			validateInsightsInfoFunctionArgs())).PrepareForEval(ctx)
+	if err != nil {
+		return nil, err
+	}
+	preparedInput := rego.EvalInput(object)
+	rs, err := query.Eval(ctx, preparedInput)
+	if err != nil {
+		return nil, err
+	}
+	return rs, nil
+}
+
+// validateInsightsInfoFunctionArgs validates the argument that would be
+// passed to our rego InsightsInfo function without executing code provided by
+// the real InsightsInfo function used elsewhere in Insights.
+// This function returns a function that is called from rego.Function().
+func validateInsightsInfoFunctionArgs() func(rego.BuiltinContext, *ast.Term) (*ast.Term, error) {
+	return func(bc rego.BuiltinContext, inf *ast.Term) (*ast.Term, error) {
+		reqInfo, err := getStringFromAST(inf)
+		if err != nil {
+			return nil, rego.NewHaltError(fmt.Errorf("unable to convert requested InsightsInfo to string: %w", err))
+		}
+		switch strings.ToLower(reqInfo) {
+		case "context", "cluster":
+			return nil, nil
```

Comment:
I think we should make `context` something like `"test"`. It might also be nice if the user could override this for the sake of testing

Reply:
I'm now passing around and processing a  `FairwindsRego.InsightsInfo` struct, the `InsightsContext` and `Cluster` members of which are configurable as flags to `validate opa`.